http_requests.py

```diff
@@ -221,3 +226,66 @@ def delete(self, url: str, return_type=None, **kwargs):
                 return_type=None,
                 **kwargs
         )
+
+
+class EncodingMethod(enum.StrEnum):
+    GZIP = 'gzip'
+
+
+def encode_request(
+    data: str | bytes | dict | typing.IO=None,
```

Comment:
should we actually allow  `dict` here (in conjunction w/ gzip)? I know `requests` allows it. However, I think, requests will translate a dict into a html-form-request IIRC (which is something we likely do not want to do here).

-> propsal: if gzip + dict: raise VE?